plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergStatistics.java

```diff
@@ -107,29 +113,42 @@ public long getSize()
         return columnSizes;
     }
 
-    public static class IcebergStatisticsBuilder
+    public static class Builder
     {
         private final Map<Integer, Type.PrimitiveType> idToTypeMapping;
         private final List<Types.NestedField> columns;
-        private final TypeManager typeManager;
         private final Map<Integer, Optional<Long>> nullCounts = new HashMap<>();
         private final Map<Integer, ColumnStatistics> columnStatistics = new HashMap<>();
         private final Map<Integer, Long> columnSizes = new HashMap<>();
+        private final Map<Type, io.trino.spi.type.Type> icebergToTrinoType;
```

Comment:
Are Iceberg Types safe to use as map keys?

(my initial thought was to have `List<> columnTrinoType` and correlate with `columns` based on list index.)